src/content/subgames.cpp

```diff
@@ -142,46 +158,42 @@ SubgameSpec findSubgame(const std::string &id)
 {
 	if (id.empty())
 		return SubgameSpec();
-	std::string share = porting::path_share;
-	std::string user = porting::path_user;
-
-	// Get games install locations
-	Strfnd search_paths(getSubgamePathEnv());
 
-	// Get all possible paths fo game
-	std::vector<GameFindPath> find_paths;
-	while (!search_paths.at_end()) {
-		std::string path = search_paths.next(PATH_DELIM);
-		path.append(DIR_DELIM).append(id);
-		find_paths.emplace_back(path, false);
-		path.append("_game");
-		find_paths.emplace_back(path, false);
-	}
+	std::string idv = normalizeGameId(id);
 
-	std::string game_base = DIR_DELIM;
-	game_base = game_base.append("games").append(DIR_DELIM).append(id);
-	std::string game_suffixed = game_base + "_game";
-	find_paths.emplace_back(user + game_suffixed, true);
-	find_paths.emplace_back(user + game_base, true);
-	find_paths.emplace_back(share + game_suffixed, false);
-	find_paths.emplace_back(share + game_base, false);
-
-	// Find game directory
-	std::string game_path;
-	bool user_game = true; // Game is in user's directory
-	for (const GameFindPath &find_path : find_paths) {
-		const std::string &try_path = find_path.path;
-		if (fs::PathExists(try_path)) {
-			game_path = try_path;
-			user_game = find_path.user_specific;
-			break;
+	GamePathMap gamepaths = getAvailableGamePaths();
+	auto found = gamepaths.find(idv);
+	if (found == gamepaths.end()) { // Failed to find the game, try to find aliased game
+		for (auto it = gamepaths.begin(); it != gamepaths.end(); ++it) {
+			const std::string conf_path = it->second.path + DIR_DELIM + "game.conf";
+			Settings conf;
+			conf.readConfigFile(conf_path.c_str());
+			if (conf.exists("aliases")) {
```

Comment:
I'd prefer to use the same code as in `getSubgameSpec` to avoid code duplication, but accepting minor performance trade-offs.
```C++
static std::unordered_set<std::string> get_aliases_from_settings(Settings &conf)
{
	std::unordered_set<std::string> aliases;
	if (!conf.exists("aliases"))
		return aliases;

	std::vector<std::string> aliases_raw = str_split(conf.get("aliases"), ',');
	for (const std::string &alias : aliases_raw)
		aliases.insert(normalizeGameId(trim(alias)));
	return aliases;
}

// ...

		Settings conf;
		conf.readConfigFile(conf_path.c_str());
		auto aliases = get_aliases_from_settings(conf);
		if (aliases.find(idv) != aliases.end()) {
			found = it;
			break;
		}
```